### ISSUE TITLE
Steam sync will not connect

### ISSUE THREAD

**Issue by Ble3azy** [gogcom/galaxy-integrations-python-api#203]
**Problem**
Connecting Steam brings up the appropriate User/Pass dialog, and successfully pings Steam for 2 factor auth, but always fails to connect with the message "offline, retry" inside the Galaxy intergrations settings. No amount of retries or restarts changes the behaviour. 
Steam and GOG are independantly working and online. Official Epic sync is working. 
Sorry I don't have an Origin Account or Ubisoft to test other community apis. 

Just built the machine, It's on a fresh windows 11 install.

Google search did yeild an old solution but the file structure appears to have changed now

Any clues? 
Thanks in advance.

**Comment by tanemahuhuta007** [comment]
same issue here

**Comment by jho728** [comment]
Same. Origin and Ubisoft Connect are fine.

**Comment by DJZK** [comment]
Same here, just 3 hours ago

**Comment by grisoustyle** [comment]
Same for me. Gives the error "Offline"

**Comment by trisk84** [comment]
I have the same issue too.....

**Comment by DJZK** [comment]
https://pastebin.com/UuX6UymD


here's the log of the Steam Integration after 1 attempt of logging in

**Comment by DonaemouS** [comment]
+1 I tried to disconnect and reconnect but it does not work, it remains offline

**Comment by marvelvang** [comment]
For further discussion see also

https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159

**Comment by fwnaw** [comment]
I have the same problem aswell

**Comment by tiagosilvaalopes** [comment]
Same issue here

**Comment by tomexx** [comment]
same issue

**Comment by skull-squadron** [comment]
Broken. [Won't be fixed since there is 0 activity on FriendsOfGalaxy.](https://github.com/FriendsOfGalaxy)

**Comment by gOOvER** [comment]
i switched to:

https://playnite.link/

GoG Galaxy was a nice Idea, but Gog is not willing to support the biggest platforms from their site

**Comment by Ble3azy** [comment]
> i switched to:
> 
> https://playnite.link/
> 
> GoG Galaxy was a nice Idea, but Gog is not willing to support the biggest platforms from their site

Yeah a already use Launchbox for my main launcher

**Comment by marvelvang** [comment]
I am also trying out playnte myself now.. after being a GOG Galaxy fan all these years since it came out.

**Comment by eldair** [comment]
Too bad galaxy got abandoned :|

**Comment by fipachu** [comment]
oh well, was fun while it lasted

**Comment by Stepan-H** [comment]
Same.

EPIC : OK
ORIGIN : OK
STEAM : BAD
UBISOFT : BAD

**Comment by Zip42** [comment]
> _UBISOFT : BAD_

Fot Ubisoft i found solution (who someone found on Reddit). It is about changing IDs on Ubisoft

- Open the Python file "%LOCALAPPDATA%\GOG.com\Galaxy\plugins\installed\uplay_afb5a69c-b2ee-4d58-b916-f4cd75d4999a\consts.py" in a text editor.
- Change line 15 and 16 to:
**_CLUB_APPID = "314d4fef-e568-454a-ae06-43e3bece12a6"
CLUB_GENOME_ID = "85c31714-0941-4876-a18d-2c7e9dce8d40"_**
- Save, connect (no need to restart Galaxy), stay connected :)

**Comment by ayesama** [comment]
> > _UBISOFT : BAD_
> 
> Fot Ubisoft i found solution (who someone found on Reddit). It is about changing IDs on Ubisoft
> 
> * Open the Python file "%LOCALAPPDATA%\GOG.com\Galaxy\plugins\installed\uplay_afb5a69c-b2ee-4d58-b916-f4cd75d4999a\consts.py" in a text editor.
> * Change line 15 and 16 to:
>   **_CLUB_APPID = "314d4fef-e568-454a-ae06-43e3bece12a6"
>   CLUB_GENOME_ID = "85c31714-0941-4876-a18d-2c7e9dce8d40"_**
> * Save, connect (no need to restart Galaxy), stay connected :)

Actually it was changed now. You can get these from the ubisoft website when you click login (you find it in the pop-up's address bar). The new ones are:

CLUB_APPID = "f35adcb5-1911-440c-b1c9-48fdc1701c68"
CLUB_GENOME_ID = "5b36b900-65d8-47f3-93c8-86bdaa48ab50"


But I can't find anything that seem to fix offline steam :(

**Comment by abelcheung** [comment]
FYI, Ubisoft connector has suffered problems long before Steam connector does. The person who was maintaining Ubisoft fix these years has also abaondoned it, and switched to Playnite.

**Comment by tomexx** [comment]
Playnite is useless to me. Every time I start it imports my 1000+ games and then downloads the metadata, which takes about 20 minutes...

**Comment by eldair** [comment]
@tomexx you should make an issue on their repo, it works fine for me

**Comment by PeacemakerZockt** [comment]
GOG Galaxy just released an update, so I had a little bit of hope, the issue might have been fixed (even though this is a community plugin). But it's still not connecting. So I think, I'll also go back to Playnite now.

**Comment by MrMody0** [comment]
> 

If anyone want to fix it, you can do it manually really easy - just swap one file and restart gog. Wait a few minutes for the steam to sync. Probably will brake after a PC restart or something and you need to login again. I am just using Hibernation and for 1day - works good. Look for the @Allexedge comments.
https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159#issuecomment-1489195530

**Comment by eldair** [comment]
There are workarounds but GOG galaxy was/is promoted with a punchline "All your games and friends in one place." and "Connect GOG GALAXY 2.0 with multiple platforms and unite all your games and friends scattered across them in one powerful app." -.- they're the one who should be fixing and maintaining all of this :| it's easy to claim all sorts of things and then let others do it for you..

**Comment by Danyelalejandro** [comment]
> There are workarounds but GOG galaxy was/is promoted with a punchline "All your games and friends in one place." and "Connect GOG GALAXY 2.0 with multiple platforms and unite all your games and friends scattered across them in one powerful app." -.- they're the one who should be fixing and maintaining all of this :| it's easy to claim all sorts of things and then let others do it for you..

I agree totally, but they are assuming as owned only the Epic and the Xbox Live integration, others are like "mods" allowed made and mantained by community

**Comment by Ble3azy** [comment]
Well, luckily theres Launchbox. Best way to manage games, emulators and collections imo. All of its intergrated accounts work and have for years. Very active community too on skins and features.

**Comment by PeacemakerZockt** [comment]
> > 
> 
> If anyone want to fix it, you can do it manually really easy - just swap one file and restart gog. Wait a few minutes for the steam to sync. Probably will brake after a PC restart or something and you need to login again. I am just using Hibernation and for 1day - works good. Look for the @Allexedge comments. [FriendsOfGalaxy/galaxy-integration-steam#159 (comment)](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159#issuecomment-1489195530)

Thank you very much. It worked for me.

**Comment by msavazzi** [comment]
steam not working for me

**Comment by v00d00m4n** [comment]
Same issue, sadly GOG galaxy does little to no progress in couple of years and does even know what is gamepad support, while Playnite actually gives gamepad supported launcher.

I think got shoulw keep core of gog galaxy for its own infrastructure and use playnite code and port current gog galaxy UI to it core and maintain its as some fork, and also help playnite to evolve. We dont need millions of launchers for other launchers and playnite seems to be the best codebase that has almost everything. It has some flaws but its much better that gog galaxy for this multi launcher purpose. It even supports emulators of all sorts and has plugins for different meta data providers, and you can set up any size and shape of arts you (feature that should be in every game launcher including steam).

**Comment by Skip75** [comment]
Since GOG no longer supports the main feature promoted by the client: the fact that it unifies the different launchers on the market (especially the biggest one, Steam), in this case I don't see the point of continuing to use GOG.

**Comment by riyando06** [comment]
i didnt undestood how to add other launcer can any one help me with this , any video ?

**Comment by Emtyloc** [comment]
I have the same problem, I was able to partially connect it by replacing the backend_steam_network.py file, but when I restart the pc it returns to the disconnected state.
Location of the file to replace -> %LOCALAPPDATA%\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8
[backend_steam_network.zip](https://github.com/gogcom/galaxy-integrations-python-api/files/11571770/backend_steam_network.zip)

**Comment by DonaemouS** [comment]
> I have the same problem, I was able to partially connect it by replacing the backend_steam_network.py file, but when I restart the pc it returns to the disconnected state. Location of the file to replace -> %LOCALAPPDATA%\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 [backend_steam_network.zip](https://github.com/gogcom/galaxy-integrations-python-api/files/11571770/backend_steam_network.zip)

Hope it could help:

- Open GoG Galaxy
- Disconnect Steam plugin
- Quit Open GoG Galaxy client (wait until is really close)
- navigate the folder user\AppData\Local\GOG.com\Galaxy\plugins\installed\
- Delete the content inside the folder steam_ca27391f-2675-49b1-92c0-896d43afa4f8
- copy the unzipped content inside the same folder
- launch the GoG Galaxy client again.
- Now try to login to Steam again

**Comment by 69NEPHILIM69** [comment]
Same here

**Comment by ijurisic** [comment]
Same issue here

**Comment by LastGodx** [comment]
A fix has been discuss and developed on this issue:
https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159
Towards the bottom you can find the latest posts and links (naturally)

Here's the repo/release for the fix taken from that issue:
https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.5-beta

It worked for me, if you'd like to go over the issue discussion to better know what you'll be installing, feel free to do so

**Comment by Emtyloc** [comment]
> A fix has been discuss and developed on this issue: [FriendsOfGalaxy/galaxy-integration-steam#159](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159) Towards the bottom you can find the latest posts and links (naturally)
> 
> Here's the repo/release for the fix taken from that issue: https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.5-beta
> 
> It worked for me, if you'd like to go over the issue discussion to better know what you'll be installing, feel free to do so

Thanks, works for me too, ABaumher is a GOD!

**Comment by ArKay74** [comment]
Fixed in May, still not installed. How is that possible?

**Comment by DontBlnkBadWolf** [comment]
> > A fix has been discuss and developed on this issue: [FriendsOfGalaxy/galaxy-integration-steam#159](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159) Towards the bottom you can find the latest posts and links (naturally)
> > Here's the repo/release for the fix taken from that issue: https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.5-beta
> > It worked for me, if you'd like to go over the issue discussion to better know what you'll be installing, feel free to do so
> 
> Thanks, works for me too, ABaumher is a GOD!

Worked for me as of right now

**Comment by Retsof123** [comment]
Worked for me, must stop and restart GOG though.

**Comment by GKid94** [comment]
![Screenshot 2023-11-26 111729](https://github.com/gogcom/galaxy-integrations-python-api/assets/18757054/63b0088a-9fc9-4706-b38b-bf8339ac6698)

**Comment by Oumbra** [comment]
Idem ici 
![image](https://github.com/gogcom/galaxy-integrations-python-api/assets/37836998/6fe7b533-7a0a-492d-a852-1393c8024cac)

**Comment by bastin-thomas** [comment]
Gog Please Fix this thing.... Add an official support for steam integration....

**Comment by QuentinSc** [comment]
<img width="630" alt="image" src="https://github.com/gogcom/galaxy-integrations-python-api/assets/16439299/c0e7ca39-0d9d-4df2-93c7-9a61a2da4720">
We're waiting...

**Comment by leonpajuelo** [comment]
> A fix has been discuss and developed on this issue: [FriendsOfGalaxy/galaxy-integration-steam#159](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159) Towards the bottom you can find the latest posts and links (naturally)
> 
> Here's the repo/release for the fix taken from that issue: https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.5-beta
> 
> It worked for me, if you'd like to go over the issue discussion to better know what you'll be installing, feel free to do so

It works for me, thanks!

**Comment by GKid94** [comment]
@mbanczerowski  Can you please fix this already we are waiting for months?

**Comment by talltechy** [comment]
broken for me as of 3/26/24

**Comment by benthiclibrarian** [comment]
Chiming in to say this is, as of a bloody year later, _still a problem._

**Comment by grayalienart** [comment]
Pretty sure this is due to Steams requirement for web sites and 3rd party launchers to have an API key now, which GOG does not have the option to enter.  LaunchBox does.    So this is partly Steams fault, although Gog has had months to implement this.

**Comment by sean8102** [comment]
Just wanted to add the Steam integration has been broken for me for a long time now as well (at least a few months). I can go through the process but then it sits at connecting for a long time before saying "offline retry". I've tried multiple times, re launching galaxy etc.

I also have Playnite installed and just checked and it's syncing with my steam account no problem.

**Comment by Nutzzz** [comment]
@sean8102 As mentioned many times above, just replace the Steam integration with [this one](https://github.com/ABaumher/galaxy-integration-steam).

**Comment by Daniel-T-Henriques** [comment]
Same here

**Comment by mariuswhr** [comment]
https://github.com/ABaumher/galaxy-integration-steam

**Comment by GKid94** [comment]
Still not working
![Screenshot 2024-11-02 190231](https://github.com/user-attachments/assets/9d909fb3-06df-482e-8cb0-9c6613f759b6)
![Screenshot 2024-11-02 190723](https://github.com/user-attachments/assets/e8be2307-ceba-4658-9719-a08b4f5e6ddc)
![Screenshot 2024-11-02 191412](https://github.com/user-attachments/assets/6dec54a1-d1ee-4c35-a0ea-aa333645ec59)

**Comment by space-gt** [comment]
DEAD?

**Comment by Danyelalejandro** [comment]
> DEAD?

yes, many plugins doesn't wok by now. Switched completely to Playnite

**Comment by rixcxardo** [comment]
@Oumbra how can I add the amazon plugin? (if you could give me a short written tutorial that would be great)
